tasks/apply_openscap_profile.yml

```diff
@@ -0,0 +1,30 @@
+---
+- name: Initialize OpenSCAP variables
+  set_fact:
+    oscap_dir: "/usr/share/xml/scap/ssg/content"
+    oscap_dist: "{{ ansible_distribution | replace('RedHat', 'rhel') | lower }}"
+    oscap_ver: "{{ ansible_distribution_major_version if ansible_distribution != 'Fedora' else '' }}"
+- name: Set OpenSCAP datastream path
+  set_fact:
+    oscap_datastream: "{{ oscap_dir }}/ssg-{{ oscap_dist }}{{ oscap_ver }}-ds.xml"
+- debug: var=oscap_datastream
+- name: Verify OpenSCAP datastream
+  stat:
+    path: "{{ oscap_datastream }}"
+  register: oscap_ds_stat
+- name: Set default OpenSCAP profile
+  shell: >-
+    oscap info --profiles {{ oscap_datastream }} |
+    grep -Ei "(standard|disa)" | sort | tail -1 | cut -d':' -f1
+  register: oscap_profile
+  when: oscap_ds_stat.stat.exists
+- debug: var=oscap_profile
+- name: Apply OpenSCAP profile
+  shell: >-
+    oscap xccdf eval --profile {{ oscap_profile.stdout }} --remediate
+    --report /root/openscap-report.html {{ oscap_datastream }} ||:
```

Comment:
I'd suggest to use command with ignore_errors: true and remove ||:

Reply:
I tried that first, but openscap is very verbose and is very likely to fail as it's applying the profile's rules in a "best effort manner", it looked messy on a small test playbook - will the stderr be displayed with `ignore_errors: true` in case of a failure ?

Reply:
If you simply add `ignore_errors: true` it will print stderr and '...ignoring' on errors but it will not fail the playbook.
If you want to avoid failing and skip any output you can add:
```
    failed_when: false
    ignore_errors: True
```

Reply:
Done, it looks fine like this as well, thanks :)